gcp/biganimal-csp-setup

```diff
@@ -0,0 +1,266 @@
+  echo "  -r, --role-name-suffix:   [Optional] The suffix part in the name of the custom role (default role name is 'biganimal-role')."
+  echo "  -h, --help:               [Optional] Show this help."
+  echo ""
+  echo "Examples:"
+  echo "    $0 -p my-sample-project-191923"
+  echo "    $0 -p my-sample-project-191923 -s \"my_sa_suffix\" -r \"my_role_suffix\""
+  echo "    this will create one service account named 'biganimal-sa-my_sa_suffix' and a custom role named 'biganimal-role-my_role_suffix'"
+  echo ""
+}
+
+check()
+{
+  #### Check gcloud CLI version
+  check_gcloud_version
+  # jq is required.
+  hash jq > /dev/null 2>&1 || { show_help; echo -e "${RED}Error: please install jq on the system${NC}"; }
+  check_project
+}
+
+version() { echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }'; }
+
+check_gcloud_version()
+{
+  fullcliversion=$(gcloud --version)
+  cliversionstripped=${fullcliversion%%$'\n'*}
+  cliversion=${cliversionstripped##* }
+  if [[ "$GOOGLE_CLOUD_SHELL" == "true" ]]; then
+    echo -e "${GREEN}Run GCP CSP setup with gcloud cli ${cliversion} in GCP CloudShell${NC}"
+  else
+    [ "$(version "$cliversion")" -lt "$(version "428.0.0")" ] && echo -e "${RED}Error: upgrade gcloud cli to 428.0.0 or later${NC}" && exit 1
+    echo -e "${GREEN}Run GCP CSP setup with gcloud cli ${cliversion}${NC}"
+  fi
+}
+
+check_project()
+{
+  if [[ "${project}" == "" ]]; then
+    echo -e "${RED}Error: missing -p, --project to specify the GCP project ID${NC}"
+    show_help
+    exit 1
+  fi
+  # check that the gcloud cli is configured to work with the given project and we have access to
+  set +e
+  current_project=$(gcloud config get-value project)
+  if [ $? -ne 0 ]; then
+    echo -e "${RED}Error while checking GCP credentials/profile configuration, please init/configure your gcloud CLI session/profile${NC}"
+    show_help
+    exit 1
+  fi
+  set -e
+  if [[ ! "${current_project}" == "$project" ]]; then
+    echo -e "${RED}Error: gcloud CLI not configured to work with the requested project $project!${NC}"
+    echo -e "${RED}The configured project ID is ${current_project}!${NC}"
+    show_help
+    exit 1
+  fi
+}
+
+while [[ $# -gt 0 ]]; do
+  key="$1"
+
+  case $key in
+    -h|--help)
+      show_help
+      exit 0
+      ;;
+    -a|--project)
+      project="$2"
+      shift 2
+      ;;
+    -p|--sa-name-suffix)
+      suffix_sa_name="$2"
+      shift 2
+      ;;
+    -r|--role-name-suffix)
+      suffix_role_name="$2"
+      shift 2
+      ;;
+    *)
+      shift
+      ;;
+  esac
+done
+
+download_dependencies()
+{
+  # a shell func to download all the dependecies of this script before starting to execute
+  # Download the latest biganimal role template or overwrite it
+  template_file=biganimal_GCP_baserole.yaml
+  curl https://raw.githubusercontent.com/EnterpriseDB/cloud-utilities/upm-15878/gcp/${template_file} -o "${CURRENT_PATH}"/${template_file}
```

Comment:
branch needs to be set to `main` after https://github.com/EnterpriseDB/cloud-utilities/pull/115 is merged